SU2_CFD/src/hybrid_RANS_LES_model.cpp

```diff
@@ -580,18 +497,21 @@ void CHybrid_Mediator::ComputeInvLengthTensor(CVariable* flow_vars,
   }
 
   switch (hybrid_res_ind) {
-  case RDELTA_INDICATOR_STRAIN_ONLY:
-    // TODO: Add rho*k contribution
-    SU2_MPI::Error("The RDELTA_INDICATOR_STRAIN_ONLY resolution adequacy option is not currently supported!", CURRENT_FUNCTION);
-    break;
-  case RDELTA_INDICATOR_FULLP:
-    SU2_MPI::Error("The RDELTA_INDICATOR_FULLP resolution adequacy option is not currently supported!", CURRENT_FUNCTION);
+  case RDELTA_INDICATOR_FULLP_WITH_DIV:
+    /*--- It's currently unclear whether the divergence of the velocity
+     * should be included in the subgrid production.  If it is included,
+     * negative eigenvalues could result, which would be interpreted as
+     * "negative" lengthscales. ---*/
     for (iDim = 0; iDim < nDim; iDim++) {
       for (jDim = 0; jDim < nDim; jDim++) {
         for (kDim = 0; kDim < nDim; kDim++) {
-          Pij[iDim][jDim] += 2.0*alpha_fac*eddy_viscosity*Sd_avg[iDim][kDim]*Om[jDim][kDim];
+          // rotation contribution
+          Pij[iDim][jDim] += 2.0*alpha_fac*eddy_viscosity*Sd_avg[iDim][kDim]*Om[kDim][jDim];
+          // Isotropic contribution (from delta_ij div_vel)
+          Pij[iDim][jDim] += 2.0*alpha_fac*eddy_viscosity*Sd_avg[iDim][jDim]*div_avg_vel/3.0;
         }
-        // TODO: Add rho*k contribution
+        // rho*k contribtuion
+	Pij[iDim][jDim] -= 2.0*alpha*rho*ktot*(Sd[iDim][jDim]+Om[iDim][jDim])/3.0;
```

Comment:
For RDELTA_INDICATOR_FULLP_WITH_DIV, isn't there a div(u) contribution to this term?

Reply:
Good catch.  I'll fix that.